src/de/error.rs

```diff
@@ -31,11 +31,14 @@ pub enum Error {
     /// The end of the BSON input was reached too soon.
     EndOfStream,
 
-    /// An invalid timestamp was encountered while decoding.
-    InvalidTimestamp(i64),
+    /// An invalid datetime was encountered while decoding.
+    InvalidDateTime {
```

Comment:
This error variant was a bit confusing, since it was referring to a datetime rather than BSON timestamp. I updated the name and included some extra information.